### PR TITLE
[Enhancement] `azurerm_image` add `disk_encryption_set_id` to `data_disk`

### PR DESCRIPTION
**Pull request by bruceharrison1984**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This adds the `disk_encryption_set_id` to the `data_disk` block

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_image` - add `disk_encryption_set_id` property to `data_disk`

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/compute/image_resource.go [hunk 1]
```diff
@@ -430,6 +437,12 @@ func expandImageDataDisks(disks []interface{}) *[]images.ImageDataDisk {
 			item.ManagedDisk = managedDisk
 		}
 
+		if id := config["disk_encryption_set_id"].(string); id != "" {
+			item.DiskEncryptionSet = &images.SubResource{
+				Id: utils.String(id),
```
Comment (stephybun) [inline internal/services/compute/image_resource.go:442]:
```suggestion
				Id: pointer.To(id),
```

File: internal/services/compute/image_resource.go [hunk 2]
```diff
@@ -513,13 +526,18 @@ func flattenImageDataDisks(input *images.ImageStorageProfile) []interface{} {
 				if disk.ManagedDisk != nil && disk.ManagedDisk.Id != nil {
 					managedDiskId = *disk.ManagedDisk.Id
 				}
+				diskEncryptionSetId := ""
+				if set := disk.DiskEncryptionSet; set != nil && set.Id != nil {
+					diskEncryptionSetId = *set.Id
```
Comment (stephybun) [inline internal/services/compute/image_resource.go:531]: We should parse the disk encryption set ID returned by the API here before setting it into state (this is so we can potentially catch any casing inconsistencies in the ID)

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @bruceharrison1984, could you take a look at the two suggestions left in-line? Once those are resolved this should be good to go.

File: internal/services/compute/image_resource.go (hunk 2 above)
Comment (bruceharrison1984) [reply internal/services/compute/image_resource.go]: Done

**Comment by bruceharrison1984** [comment]
I'm not sure what the `Breaking Schema Changes` test is, and why it is suddenly failing?

File: internal/services/compute/image_resource.go [hunk 3]
```diff
@@ -463,7 +476,8 @@ func flattenImageOSDisk(input *images.ImageStorageProfile) []interface{} {
 			}
 			diskEncryptionSetId := ""
 			if set := v.DiskEncryptionSet; set != nil && set.Id != nil {
-				diskEncryptionSetId = *set.Id
+				encryptionId, _ := commonids.ParseDiskEncryptionSetIDInsensitively(*set.Id)
+				diskEncryptionSetId = encryptionId.String()
```
Comment (stephybun) [inline internal/services/compute/image_resource.go:480]: `.String()` outputs the ID information in a human-readable way for error and log messages, these probably want to use `.ID()`
```suggestion
				diskEncryptionSetId = encryptionId.ID()
```

File: internal/services/compute/image_resource.go [hunk 4]
```diff
@@ -513,13 +527,19 @@ func flattenImageDataDisks(input *images.ImageStorageProfile) []interface{} {
 				if disk.ManagedDisk != nil && disk.ManagedDisk.Id != nil {
 					managedDiskId = *disk.ManagedDisk.Id
 				}
+				diskEncryptionSetId := ""
+				if set := disk.DiskEncryptionSet; set != nil && set.Id != nil {
+					encryptionId, _ := commonids.ParseDiskEncryptionSetIDInsensitively(*set.Id)
+					diskEncryptionSetId = encryptionId.String()
```
Comment (stephybun) [inline internal/services/compute/image_resource.go:533]:
```suggestion
					diskEncryptionSetId = encryptionId.ID()
```

**Review by stephybun — CHANGES_REQUESTED** [review]
No need to worry about the breaking change check, I believe that's going to be removed soon/someone will take a look

**Review by stephybun — APPROVED** [review]
Thanks @bruceharrison1984! LGTM 👍

**Pull request merged**